File: tool/Database initializer/PythonSpotify_v2/pathFormatter.py
import platform
import os

class PathFormatter():

    @staticmethod
    def formatPath(path: str):
        systemType = platform.system()

        if systemType == "Windows":
            return path.replace("/", "\\")
        elif systemType == "Linux":
            return path.replace("\\", "/")
    
    @staticmethod
    def getSplitterChar() -> str:
        systemType = platform.system()

        if systemType == "Windows":
            return "\\"
        elif systemType == "Linux":
            return "/"
        

File: server/src/main/java/utility/PathFormatter.java
package utility;

import java.util.Locale;


public final class PathFormatter 
{
    private static String OS = System.getProperty("os.name", "unknown").toLowerCase(Locale.ROOT);
    private static String pathSeparator = System.getProperty("path.separator");

    public static String formatPath(String path) 
    {
        //System.out.println(OS);
        
        if (OS.contains("win")) {
            return path.replace("/", "\\");
        }
        else if (OS.contains("nux")) {
            return path.replace("\\", "/");
        }
        else if (OS.contains("mac")) {
            return path.replace("\\", "/");
        }
        else {
            return path;
        }
    } 

    public static String getPathSeparator() {
        if (OS.contains("win")) {
            return "\\";
        }
        else if (OS.contains("nux")) {
            return "/";
        }
        else if (OS.contains("mac")) {
            return "/";
        }
        else {
            return null;
        }
    }
}
